upgrade/v1beta1/upgrade.go

```diff
@@ -120,6 +140,49 @@ func upgradeVolumes(namespace string, lhClient *lhclientset.Clientset) error {
 	return nil
 }
 
+func fixupVolumes(namespace string, lhClient *lhclientset.Clientset) error {
+	volumes, err := lhClient.LonghornV1beta1().Volumes(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: getLabelSelectorNotEqualV1beta2APIVersion(),
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range volumes.Items {
+		existing := obj.DeepCopy()
+
+		if obj.Spec.DiskSelector == nil {
+			obj.Spec.DiskSelector = []string{}
+		}
+		if obj.Spec.NodeSelector == nil {
+			obj.Spec.NodeSelector = []string{}
+		}
+		if obj.Spec.RecurringJobs == nil {
+			obj.Spec.RecurringJobs = make([]longhornV1beta1.VolumeRecurringJobSpec, 0)
+		}
+		for i, src := range obj.Spec.RecurringJobs {
+			dst := longhornV1beta1.VolumeRecurringJobSpec{}
+			if err := copier.Copy(&dst, &src); err != nil {
+				return err
+			}
+			if dst.Groups == nil {
+				dst.Groups = []string{}
+			}
+			if dst.Labels == nil {
+				dst.Labels = make(map[string]string, 0)
+			}
+			obj.Spec.RecurringJobs[i] = dst
+		}
```

Comment:
Maybe you can just reuse `initXXX` here instead of repeating the implementation, but you need to make `initXXX` exported first. WDYT?

Reply:
Agree, it's better to reuse the functions, but the `datastore/longhorn.go` and `upgrade/v1beta1/upgrade.go` handle `v1beta2` and `v1beta2` objects, respectively.
Seems no simple way to handle them into one function. In addition, `v1beta1` objects' fields are not changed anymore, but the fields in `v1beta2` might be changed in the future. May handling them be better separately?

Reply:
I think what you want to say is `datastore/longhorn.go` is handling `v1beta1`.  Ok, it seems like a limitation for now except we have another reflection layer to know which type to handle but it will be an over engineering. 

Let's keep this as is.

Reply:
Yes, sorry for the typo.